tdrs-backend/tdpservice/reports/views.py

```diff
@@ -18,24 +22,42 @@ class ReportFileViewSet(ModelViewSet):
 
     http_method_names = ['get', 'post', 'head']
     parser_classes = [MultiPartParser]
-    permission_classes = [CanUploadReport]
+    permission_classes = [ReportFilePermissions]
     serializer_class = ReportFileSerializer
     queryset = ReportFile.objects.all()
 
+    @action(methods=["get"], detail=True)
+    def download(self, request, pk=None):
+        """Retrieve a file from s3 then stream it to the client."""
+        record = self.get_object()
+
+        response = StreamingHttpResponse(
+            FileWrapper(record.file),
+            content_type='txt/plain'
+        )
+        file_name = record.original_filename
+        response['Content-Disposition'] = f'attachment; filename="{file_name}"'
+        return response
+
 
 class GetYearList(APIView):
     """Get list of years for which there are reports."""
 
     query_string = False
     pattern_name = "report-list"
-    permission_classes = [CanDownloadReport]
+    permission_classes = [ReportFilePermissions]
 
-    def get(self, request, **kargs):
+    def get(self, request, **kwargs):
         """Handle get action for get list of years there are reports."""
         user = request.user
         is_ofa_admin = user.groups.filter(name="OFA Admin").exists()
 
-        stt_id = kargs.get('stt') if is_ofa_admin else user.stt.id
+        stt_id = kwargs.get('stt') if is_ofa_admin else user.stt.id
+        if not stt_id:
+            return Response(
+                {'detail': 'Must supply a valid STT'},
+                status=HTTP_400_BAD_REQUEST
```

Comment:
This is a more fitting response for when an OFA admin doesn't supply an STT id and allowed consolidation of the two permissions classes revolving around ReportFiles (which were almost identical).

Reply:
> > * not having any success downloading files for my stt as a data prepper.
> 
> @ADPennington it looks like you're using the STT id to attempt to download the report file. In that case it goes off the ID of the ReportFile record itself (`id` in the response from creating a report).

confirmed! thank you @jtwillis92 ! fixed this and resolved the data prepper downloading (example below for GA):
<img width="908" alt="833p10" src="https://user-images.githubusercontent.com/63075587/124300390-85840d80-db2c-11eb-9a8b-74e965405117.png">